torch/jit/__init__.py

```diff
@@ -28,6 +28,25 @@ def __repr__(self):
 VOLATILE = Placeholder("VOLATILE")
 
 
+def get_tracing_state(*args):
+    if not torch._C.is_tracing(args):
+        return None
+    return torch._C.get_tracing_state(args)
+
+
+@contextlib.contextmanager
+def scope(scope_name, vars):
+    if not torch._C.is_tracing(vars):
+        tracing_state = None
+    else:
+        tracing_state = torch._C.get_tracing_state(vars)
+    if tracing_state:
+        tracing_state.push_scope(scope_name)
+    yield
```

Comment:
I think it might be good to wrap this in a `try`/`finally`, so that if errors occur we still properly clean up the `tracing_state`.
Maybe something like
```python
try:
    yield
finally:
    if tracing_state:
        tracing_state.pop_scope()
```